### ISSUE TITLE
False positive for 'variable is mutated but never accesed'

### ISSUE THREAD

**Issue by alek13** [lunarmodules/luacheck#110]
Relates to https://github.com/mpeterv/luacheck/issues/72
Was fixed: 1e62378af7d4f734dd262b77bbdd7329d75ac8ad

But still false-positive for this case:
```lua
local test_table = {}

test_table["some_sting_key_or_var_with_key_name"] = "some_value"
```

**Additional info:**
```console
$ luacheck --version
Luacheck: 1.1.2
Lua: PUC-Rio Lua 5.1
Argparse: 0.7.1
LuaFileSystem: 1.8.0
LuaLanes: Not found
```

**Comment by alerque** [comment]
I don't see how this is a false positive. It looks like a true positive to me. The case that was fixed is completely different because of the scope. Your example creates a local variable and then never uses it. Since Luacheck can confirm there are no references to it in the given scope it correctly throws a lint warning.

Any use of the table should clear this, e.g.:

```
local test_table = {}

test_table["some_sting_key_or_var_with_key_name"] = "some_value"

print(test_table)
```

**Comment by alek13** [comment]
sorry, my bad.
anyway thanks for your time
